plasmapy/diagnostics/langmuir.py

```diff
@@ -1,153 +1,1073 @@
+                              can_be_inf=False, can_be_nan=True)
+
+    def get_padded_limit(self, padding, log=False):
+        r"""Return the limits of the current range for plotting, taking into
+        account padding. Matplotlib lacks this functionality.
+
+        Parameters
+        ----------
+        padding : float
+            The padding ratio as a float between 0.0 and 1.0.
+
+        log : boolean, optional
+            If True the calculation will be performed on a logarithmic scale.
+            Default is False.
+
+        """
+
+        ymax = np.max(self.current).to(u.A).value
+
+        if(log):
+            ymin = np.min(np.abs(
+                    self.current[self.current != 0])).to(u.A).value
+            return [ymin * 10**(-padding * np.log10(ymax / ymin)),
+                    ymax * 10**(padding * np.log10(ymax / ymin))] * u.A
+        else:
+            ymin = np.min(self.current).to(u.A).value
+            return [ymin - padding * (ymax - ymin),
+                    ymax + padding * (ymax - ymin)] * u.A
+
+    def plot(self):
+        r"""Plot the characteristic in matplotlib."""
+
+        with quantity_support():
+            plt.figure()
+            plt.scatter(self.bias, self.current,
+                        marker='.', color='k')
+            plt.title("Probe characteristic")
+
+
+@utils.check_quantity({'probe_area': {'units': u.m**2,
+                                      'can_be_negative': False,
+                                      'can_be_complex': False,
+                                      'can_be_inf': False,
+                                      'can_be_nan': False}})
+def swept_probe_analysis(probe_characteristic, probe_area, gas,
+                         bimaxwellian=False, visualize=False,
+                         plot_electron_fit=False, plot_EEDF=False):
+    r"""Attempt to perform a basic swept probe analysis based on the provided
+    characteristic and probe data. Suitable for single cylindrical probes in
+    low-pressure DC plasmas, since OML is applied.
 
     Parameters
     ----------
-    potential_sweep : ~astropy.units.Quantity
-        The swept electric potential applied to the probe in units
-        convertible to V.
+    probe_characteristic : Characteristic
+        The swept probe characteristic that is to be analyzed.
+
+    probe_area : Quantity
+        The area of the probe exposed to plasma in units convertible to m^2.
+
+    gas : float
+        The (mean) mass of the background gas in atomic mass units (amu).
+
+    visualize : bool, Optional
+        Can be used to plot the characteristic and the obtained parameters.
+        Default is False.
+
+    plot_electron_fit : bool, Optional
+        If True, the fit of the electron current in the exponential section is
+        shown. Default is False.
+
+    plot_EEDF : bool, Optional
+        If True, the EEDF is computed and shown. Default is False.
+
+    Returns
+    -------
```

Comment:
Maybe we should add something like a sentence saying "A dictionary with the following entries:"?